lib/dpl/provider/deis.rb

```diff
@@ -44,6 +44,31 @@ def setup_git_ssh(path, key_path)
         unless context.shell "./deis git:remote --app=#{option(:app)}"
           error 'Adding git remote failed.'
         end
+
+        wait_for_git_access
+      end
+
+      def wait_for_git_access()
+        retry_count=0
+        max_retries=30
+
+        #Parse out the git remote host and port
+        git_remote=`cat .git/config  | grep -A1 '[remote] "deis"' | tail -n 1 | sed 's/.*url = //'`
+        remote_uri=git_remote.split("ssh://")[1].split("/")[0]
```

Comment:
This assumes that `git_remote` is nonempty; which might be true in practice (that there is a remote named `deis`), but as you can see in the failing specs, it may not be true.